packages/app/src/cli/services/function/replay.test.ts

```diff
@@ -0,0 +1,134 @@
+import {FunctionRunData, replay} from './replay.js'
+import {runFunctionRunner} from './build.js'
+import {testApp, testFunctionExtension, testDeveloperPlatformClient} from '../../models/app/app.test-data.js'
+import {ensureConnectedAppFunctionContext} from '../generate-schema.js'
+import {selectFunctionRunPrompt} from '../../prompts/function/replay.js'
+import {inTemporaryDirectory, mkdir, writeFile} from '@shopify/cli-kit/node/fs'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {getLogsDir} from '@shopify/cli-kit/node/logs'
+import {describe, expect, test, vi} from 'vitest'
+import {exec} from '@shopify/cli-kit/node/system'
+
+vi.mock('../generate-schema.js')
+vi.mock('../../prompts/function/replay.js')
+vi.mock('@shopify/cli-kit/node/logs')
+vi.mock('./build.js')
+vi.mock('@shopify/cli-kit/node/system')
+
+const RUN1: FunctionRunData = {
+  shop_id: 69665030382,
+  api_client_id: 124042444801,
+  payload: {
+    input: '{}',
+    input_bytes: 136,
+    output: '{}',
+    output_bytes: 195,
+    function_id: '34236fa9-42f5-4bb6-adaf-956e12fff0b0',
+    logs: '',
+    fuel_consumed: 458206,
+  },
+  event_type: 'function_run',
+  cursor: '2024-05-31T15:29:47.291530Z',
+  status: 'success',
+  log_timestamp: '2024-05-31T15:29:46.741270Z',
+  identifier: 'abcdef',
+}
+const RUN1_FILENAME = RUN1.log_timestamp.replace(/:/g, '_')
+
+const RUN2: FunctionRunData = {
+  shop_id: 69665030382,
+  api_client_id: 124042444801,
+  payload: {
+    input: '{}',
+    input_bytes: 136,
+    output: '{}',
+    output_bytes: 195,
+    function_id: '34236fa9-42f5-4bb6-adaf-956e12fff0b0',
+    logs: '',
+    fuel_consumed: 458206,
+  },
+  event_type: 'function_run',
+  cursor: '2024-05-31T15:29:47.291530Z',
+  status: 'success',
+  log_timestamp: '2024-05-31T15:29:50.741270Z',
+  identifier: '123456',
+}
+const RUN2_FILENAME = RUN2.log_timestamp.replace(/:/g, '_')
+
+describe('replay', () => {
+  test('reads the app log directory, parses the function runs, and invokes function-runner aa', async () => {
```

Comment:
Should we add a test for the situation where the app log directory is empty?

Reply:
Good callout! 
I see the other prompts return undefined in this scenario so I'll do that. 
Then here I'll throw an error showing the directory that we tried to find runs in.